shuup/xtheme/resources.py

```diff
@@ -213,3 +250,49 @@ def add_resource(context, location, resource):
     if rc:
         return bool(rc.add_resource(location, resource))
     return False
+
+
+def valid_view(context):
+    """
+    Prevent adding the global snippet in admin views and in editor view
+    """
+    view_class = getattr(context["view"], "__class__", None) if context.get("view") else None
+    if not view_class or not context.get("request"):
+        return False
+
+    request = context.get("request")
+    if request:
+        match = request.resolver_match
+        if match and match.app_name == "shuup_admin":
+            return False
+
+    view_name = getattr(view_class, "__name__", "")
+    if view_name == "EditorView":
+        return False
+
+    return True
+
+
+def inject_global_snippet(context, content):
+    if not valid_view(context):
+        return
+
+    from shuup.xtheme import get_current_theme
+    from shuup.xtheme.models import Snippet, SnippetType
+    shop = get_shop(context["request"])
+
+    for snippet in Snippet.objects.filter(shop=shop):
```

Comment:
This likely needs to be cached. We can't loop over snippets on every page load.

Reply:
true